godot-core/src/registry/class.rs

```diff
@@ -331,21 +331,29 @@ pub(crate) fn try_dynify_object<T: GodotClass, D: ?Sized + 'static>(
 /// Responsible for creating hint_string for [`DynGd<T, D>`][crate::obj::DynGd] properties which works with [`PropertyHint::NODE_TYPE`][crate::global::PropertyHint::NODE_TYPE] or [`PropertyHint::RESOURCE_TYPE`][crate::global::PropertyHint::RESOURCE_TYPE].
 ///
 /// Godot offers very limited capabilities when it comes to validating properties in the editor if given class isn't a tool.
-/// Proper hint string combined with `PropertyHint::NODE_TYPE` or `PropertyHint::RESOURCE_TYPE` allows to limit selection only to valid classes - those registered as implementors of given `DynGd<T, D>`'s `D` trait.
+/// Proper hint string combined with `PropertyHint::RESOURCE_TYPE` allows to limit selection only to valid classes - those registered as implementors of given `DynGd<T, D>`'s `D` trait.
+/// Godot editor allows to export only one node type with `PropertyHint::NODE_TYPE` – therefore we are returning only the base class.
 ///
 /// See also [Godot docs for PropertyHint](https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#enum-globalscope-propertyhint).
-pub(crate) fn get_dyn_property_hint_string<D>() -> GString
+pub(crate) fn get_dyn_property_hint_string<T, D>() -> String
 where
+    T: GodotClass,
     D: ?Sized + 'static,
 {
+    // Exporting multiple node types is not supported.
+    if T::inherits::<classes::Node>() {
```

Comment:
I'm not a fan of these multiple T::inherits:: checks when we are exporting Array<DynGd<...>> but we are doing so only in the editor, so it should be fine?

Reply:
To understand, what is your concern here? That `T::inherits::<classes::Node>()` is a runtime check as opposed to trait static dispatch?

Because I would expect the compiler to be reasonably good at optimizing this, as all is known at compile time :thinking:

Reply:
Yeah, it's just much nicer to write procedural code than factor this out into multiple traits. The latter is often necessary because of bounds or because code wouldn't compile otherwise; not so here.

Same reason why C++ added [`if constexpr`](https://en.cppreference.com/w/cpp/language/if) (aka "static if").